powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/repository/OperationRepository.java

```diff
@@ -61,4 +64,15 @@
             "ORDER BY o.timestampCreated")
     Stream<OperationEntity> findExpiredPendingOperations(Date timestamp);
 
+    @Query("SELECT o.id FROM OperationEntity o " +
+            "WHERE o.timestampExpires < :timestamp AND o.status = io.getlime.security.powerauth.app.server.database.model.OperationStatusDo.PENDING ")
+    List<String> findExpiredPendingOperationIds(Date timestamp);
+
+    @Modifying( clearAutomatically = true)
+    @Query("UPDATE OperationEntity o SET status = io.getlime.security.powerauth.app.server.database.model.OperationStatusDo.EXPIRED " +
+            "WHERE o.id IN (:ids)")
```

Comment:
The `IN` operator has maximum number of parameters of 1000 in Oracle. For other databases there is a general limit for the number of parameters present. We should make sure this query works correctly even when the number of parameters is large, which can happen quite easily.